src/app-layout/__tests__/trigger-button.test.tsx

```diff
@@ -0,0 +1,357 @@
+describe('Visual Refresh Toolbar trigger-button', () => {
+  beforeEach(() => jest.clearAllMocks());
+
+  test('renders correctly with without badge', async () => {
+    const ref = React.createRef<React.Ref<ButtonProps.Ref>>();
+    const { wrapper, getByTestId } = await renderVisualRefreshToolbarTriggerButton(
+      {
+        badge: false,
+      },
+      ref as any
+    );
+
+    expect(wrapper).not.toBeNull();
+    expect(getByTestId(mockTestId)).toBeTruthy();
+    const button = wrapper.find('button');
+    expect(button).toBeTruthy();
+    expect(button!.getElement().getAttribute('aria-label')).toEqual(mockProps.ariaLabel);
+    expect(wrapper!.findIcon()).toBeTruthy();
+    expect(wrapper.findByClassName(toolbarTriggerButtonStyles.dot)).toBeNull();
+    expect(wrapper.findBadge()).toBeNull();
+  });
+
+  test('renders correctly with iconName', async () => {
+    const ref = React.createRef<React.Ref<ButtonProps.Ref>>();
+    const { wrapper, getByTestId } = await renderVisualRefreshToolbarTriggerButton({}, ref as any);
+
+    expect(wrapper).not.toBeNull();
+    const button = wrapper.find('button');
+    expect(getByTestId(mockTestId)).toBeTruthy();
+    expect(button).toBeTruthy();
+    expect(button!.getElement().getAttribute('aria-label')).toEqual(mockProps.ariaLabel);
+    expect(wrapper!.findIcon()).toBeTruthy();
+  });
+
+  test('renders correctly with iconSvg', async () => {
+    const iconTestId = 'icon-test-id';
+    const icon = (
+      <svg data-testid={iconTestId} viewBox="0 0 24 24">
+        <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z" />
+      </svg>
+    );
+    const ref = React.createRef<React.Ref<ButtonProps.Ref>>();
+    const { wrapper, getByTestId } = await renderVisualRefreshToolbarTriggerButton(
+      {
+        iconSvg: icon,
+      },
+      ref as any
+    );
+
+    expect(wrapper).not.toBeNull();
+    const button = wrapper.find('button');
+    expect(getByTestId(mockTestId)).toBeTruthy();
+    expect(button).toBeTruthy();
+    expect(button!.getElement().getAttribute('aria-label')).toEqual(mockProps.ariaLabel);
+    expect(wrapper!.findIcon()).toBeTruthy();
+    expect(getByTestId(iconTestId)).toBeTruthy();
+  });
+
+  test.each([true, false])('renders correctly and with when disabled is %s', async disabledValue => {
+    const ref = React.createRef<React.Ref<ButtonProps.Ref>>();
+    const mockClickSpy = jest.fn();
+    const { wrapper, getByTestId } = await renderVisualRefreshToolbarTriggerButton(
+      {
+        disabled: disabledValue,
+        onClick: mockClickSpy,
+      },
+      ref as any
+    );
+    expect(wrapper).not.toBeNull();
+    const button = wrapper.find('button')!;
+    expect(getByTestId(mockTestId)).toBeTruthy();
+    button.click();
+    expect(mockClickSpy).toBeCalledTimes(disabledValue ? 0 : 1);
+  });
+
+  test('renders an empty button when no iconName and iconSVG prop', async () => {
```

Comment:
would it be nicer to fallback to some icon instead?

Reply:
This can be added in a followup